storage/google/src/server/wrappers/StorageControlClientWrapper.ts

```diff
@@ -0,0 +1,175 @@
+    const managedFolder = this.managedFolderPath(folderName);
+    try {
+      await this._client.deleteManagedFolder({
+        name: managedFolder,
+      });
+    } catch (error) {
+      if (isGoogleError(error) && error.code != 5) {
+        throw error;
+      }
+    }
+  }
+
+  public async getManagedFoldersNextPage(options: {
+    maxPageSize: number;
+    continuationToken?: string;
+  }): Promise<EntityCollectionPage<BaseDirectory>> {
+    const parent = this.bucketPath;
+    const [folders, _, response] = await this._client.listManagedFolders(
+      {
+        parent,
+        pageSize: options.maxPageSize,
+        pageToken: options.continuationToken,
+      },
+      { autoPaginate: false }
+    );
+    const directories: BaseDirectory[] =
+      folders.map((entry) => {
+        return {
+          baseDirectory: entry.name?.split("/").slice(5, -1).join("/"),
+        } as BaseDirectory;
+      }) ?? [];
+    const continuationToken = response?.nextPageToken;
+    const page: EntityCollectionPage<BaseDirectory> = {
+      entities: directories,
+      next: !continuationToken
+        ? undefined
+        : () =>
+            this.getManagedFoldersNextPage({
+              maxPageSize: options.maxPageSize,
+              continuationToken: continuationToken,
+            }),
+    };
+    return page;
+  }
+
+  public async getManagedFolders(): Promise<BaseDirectory[]> {
+    const parent = this._client.bucketPath("_", this._config.bucketName);
+    const [folders, _1, _2] = await this._client.listManagedFolders({
+      parent,
+    });
+    const directories: BaseDirectory[] =
+      folders.map(
+        (entry) => ({ baseDirectory: entry.name } as BaseDirectory)
+      ) ?? [];
+    return directories;
+  }
+
+  public async managedFolderExists(folderName: string): Promise<boolean> {
+    const managedFolder = this.managedFolderPath(folderName);
+    try {
+      await this._client.getManagedFolder({
+        name: managedFolder,
+      });
+      return true;
+    } catch (error) {
+      if (isGoogleError(error) && error.code == 5) {
+        return false;
+      }
+      throw error;
+    }
+  }
+
+  public async createAccessToken(
+    action: GoogleStorageConfigType,
+    folderName: string
+  ): Promise<GoogleTransferConfig> {
+    const cab = {
+      accessBoundary: {
+        accessBoundaryRules: [
+          {
+            availableResource: this.bucketUri,
+            availablePermissions: [roleFromConfigType(action)],
+            availabilityCondition: {
+              expression:
+                `resource.name.startsWith('${this.bucketPath}/objects/${folderName}') || ` +
+                `api.getAttribute('storage.googleapis.com/objectListPrefix', '')` +
+                `.startsWith('${folderName}/')`,
+            },
+          },
+        ],
+      },
+    };
+    const googleAuth = new GoogleAuth({
+      scopes: "https://www.googleapis.com/auth/cloud-platform",
```

Comment:
Do we plan to eventually have our own scope so we can have control over token expiration times?

Reply:
I agree, an hour is good for now. Just curious if we had looked into it, as custom expiration times are supported by azure access tokens.